mesh/mesh.hpp

```diff
@@ -978,16 +988,31 @@ class Mesh
 
        The IntegrationRule used with GetFaceGeometricFactors needs to remain
        valid until the internally stored FaceGeometricFactors objects are
-       destroyed (by either calling Mesh::DeleteGeometricFactors or the Mesh
-       destructor). */
-   const FaceGeometricFactors* GetFaceGeometricFactors(const IntegrationRule& ir,
-                                                       const int flags,
-                                                       FaceType type,
-                                                       MemoryType d_mt = MemoryType::DEFAULT);
+       destroyed (by either calling Mesh::DeleteGeometricFactors(),
+       Mesh::NodesUpdated(), or the Mesh destructor).
+
+       If the device MemoryType parameter @a d_mt is specified, then the
+       returned object will use that type unless it was previously allocated
+       with a different type.
+
+       The returned pointer points to an internal object that may be invalidated
+       by mesh operations such as refinement, vertex/node movement, etc. Since
+       not all such modifications can be tracked by the Mesh class (e.g. when
+       using the pointer returned by GetNodes() to change the nodes) one needs
+       to account for such changes by calling the method NodesUpdated() which,
+       in particular, will call DeleteGeometricFactors(). */
+   const FaceGeometricFactors* GetFaceGeometricFactors(
```

Comment:
Why not add a check to see if things are still valid and if not perform the update internally.

Reply:
That would be ideal, but there isn't really a way to check that things haven't been invalidated. That is what this sentence is trying to convey: "Since not all such modifications can be tracked by the Mesh class (e.g. when using the pointer returned by GetNodes() to change the nodes)"

Reply:
@mlstowell [originally suggested](https://github.com/mfem/mfem/issues/3042#issuecomment-1147645568) calling `DeleteGeometricFactors()` in `Transform()`, which is an important detail this PR needs to add. 

Mitigating the underlying problem is preferable to just documenting it.

Reply:
Added in 44380ab02.